compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -398,6 +404,8 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
                 // avoid type aliases for tuples
                 Right(MirrorSource.GenericTuple(types))
               case _ => reduce(tp.underlying)
+          case tp @ AppliedType(tref: TypeRef, List(_, typeTuple)) if tref == defn.NamedTupleTypeRef =>
+            Right(MirrorSource.NamedTuple(tp.namedTupleElementTypes(false)))
```

Comment:
what changes with `derived=true` instead of `false`?

Reply:
`derived=true` allows to convert union and intersection types of NamedTuples into NamedTuple `AppliedType` types. We already know it's a `NamedTupleDirect` from the unapply above, so we don't need to do that (and setting `true` would not change anything here).